### PR TITLE
Target latest API + Material Design

### PR DESCRIPTION
**Pull request by jaredsburrows**
Please see https://github.com/passy/Android-DirectoryChooser/issues/54

 - Update Robolectric to 3.0-rc3 - :heavy_check_mark:
 - AppCompatv7 - Material Design - :heavy_check_mark:
 - Use `testCompile` for unit tests - :heavy_check_mark:
 - Remove Robolectric plugin - not needed - :heavy_check_mark:
 - Update Android gradle plugin to 1.2.3 - needed for unit tests - :heavy_check_mark:

You will **now** see `Material themed Fragment` and `Activity`.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
     }
-  }
 
-  dependencies {
-    classpath 'com.android.tools.build:gradle:1.0.1'
-    classpath 'org.robolectric:robolectric-gradle-plugin:0.14.+'
-  }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:1.2.3'
+    }
```
Comment (jaredsburrows) [inline build.gradle]: You do not need the Robolectric plugin.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Sun Apr 06 15:14:44 BST 2014
+#Tue Jul 07 20:20:14 EDT 2015
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=http\://services.gradle.org/distributions/gradle-2.2-all.zip
+distributionUrl=http\://services.gradle.org/distributions/gradle-2.4-bin.zip
```
Comment (jaredsburrows) [inline gradle/wrapper/gradle-wrapper.properties]: Download via HTTP.

File: gradlew.bat [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+@rem Setup the command line
+
+set CLASSPATH=%APP_HOME%\gradle\wrapper\gradle-wrapper.jar
+
+@rem Execute Gradle
+"%JAVA_EXE%" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %GRADLE_OPTS% "-Dorg.gradle.appname=%APP_BASE_NAME%" -classpath "%CLASSPATH%" org.gradle.wrapper.GradleWrapperMain %CMD_LINE_ARGS%
+
+:end
+@rem End local scope for the variables with windows NT shell
+if "%ERRORLEVEL%"=="0" goto mainEnd
```
Comment (jaredsburrows) [inline gradlew.bat]: Updated to Gradle 2.4

File: library/build.gradle [hunk 4]
```diff
@@ -8,12 +7,18 @@ repositories {
...
-  androidTestCompile 'org.mockito:mockito-all:1.9.+'
+    compile 'com.android.support:appcompat-v7:22.2.0'
+    compile 'com.gu:option:1.3'
+    testCompile 'junit:junit:4.12'
+    testCompile 'org.mockito:mockito-all:1.9.5'
+    testCompile('com.squareup:fest-android:1.0.8') {
+        exclude group: 'com.android.support', module: 'support-v4'
+    }
+    testCompile 'org.apache.maven:maven-ant-tasks:2.1.3' // fixes issue on linux/mac
+    testCompile("org.robolectric:shadows-support-v4:3.0-rc3") {
```
Comment (jaredsburrows) [inline library/build.gradle]: Updated to Robolectric 3.0.

File: library/build.gradle [hunk 5]
```diff
@@ -22,7 +27,7 @@ android {
 
   defaultConfig {
     minSdkVersion 11
-    targetSdkVersion 18
+    targetSdkVersion 22
```
Comment (jaredsburrows) [inline library/build.gradle]: Target latest.

File: library/build.gradle [hunk 6]
```diff
@@ -61,30 +57,3 @@ gradle.taskGraph.whenReady { taskGraph ->
     allprojects { ext."signing.password" = console.readPassword("PGP Private Key Password: ") }
   }
 }
-
```
Comment (jaredsburrows) [inline library/build.gradle]: Latest Android Plugin supports unit tests.

File: library/src/main/java/net/rdrei/android/dirchooser/DirectoryChooserActivity.java [hunk 7]
```diff
@@ -2,19 +2,19 @@
...
 import android.content.Intent;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
 import android.view.MenuItem;
 
 /**
  * Let's the user choose a directory on the storage device. The selected folder
  * will be sent back to the starting activity as an activity result.
```
Comment (jaredsburrows) [inline library/src/main/java/net/rdrei/android/dirchooser/DirectoryChooserActivity.java]: Material Design.

File: library/src/main/res/values/styles.xml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="DirectoryChooserTheme" parent="Theme.AppCompat.Light">
```
Comment (jaredsburrows) [inline library/src/main/res/values/styles.xml]: Material Design.

File: sample/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -8,21 +8,22 @@
         android:allowBackup="true"
         android:icon="@drawable/ic_launcher"
         android:label="@string/app_name"
-        android:theme="@android:style/Theme.Holo.Light" >
-        <activity android:name="net.rdrei.android.dirchooser.sample.DirChooserSample" >
+        android:theme="@style/DirectoryChooserTheme">
```
Comment (jaredsburrows) [inline sample/src/main/AndroidManifest.xml]: Apply App Compat theme for Material Design.

**Comment by jaredsburrows** [comment]
@passy Let me also update TravisCI.

**Comment by jaredsburrows** [comment]
@passy  Check out the "sample" to test out the Material directory chooser.

**Comment by jaredsburrows** [comment]
@passy Have you tested this out?

File: .travis.yml [hunk 10]
```diff
@@ -1,17 +1,28 @@
...
 
 script:
-  - TERM=dumb ./gradlew testDebug --info
+ - ./gradlew clean assembleDebug
+
+after_success:
+ - ./gradlew testDebug --info
+
+notifications:
+  email: false
```
Comment (passy) [inline .travis.yml]: You can leave this on. I don't mind getting emails if it breaks. :)

File: gradlew.bat [hunk 11]
```diff
@@ -0,0 +1,90 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (passy) [inline gradlew.bat]: Can you remove this? I don't support Windows and if someone wants to try it there, they can manually create the file.

File: library/build.gradle (hunk 4 above)
Comment (passy) [inline library/build.gradle]: Nice!

File: library/build.gradle (hunk 5 above)
Comment (passy) [inline library/build.gradle]: Woo, finally supported!

**Comment by passy** [comment]
Thanks so much for the update, this is great work! :)

File: gradlew.bat (hunk 11 above)
Comment (jaredsburrows) [inline gradlew.bat]: Yes.

File: .travis.yml (hunk 10 above)
Comment (jaredsburrows) [inline .travis.yml]: Sounds good.

File: library/build.gradle (hunk 4 above)
Comment (jaredsburrows) [inline library/build.gradle]: Robolectric 3.0, release came up, so I will make a slight up date.

**Comment by jaredsburrows** [comment]
Thank you!

File: gradlew.bat (hunk 3 above)
Comment (jaredsburrows) [inline gradlew.bat]: Gradle 2.5 just came out, I will update.

**Comment by jaredsburrows** [comment]
@passy Can we merge?

**Comment by passy** [comment]
We can. Thanks for your work. Much appreciated! :)

**Pull request merged**